test/LocatedTriplesTest.cpp

```diff
@@ -0,0 +1,629 @@
+    });
+    auto locatedTriplesPerBlock = makeLocatedTriplesPerBlock({
+        LT{1, IT(1, 10, 10), false},  // Delete row 0
+        LT{1, IT(1, 10, 11), true},   // Insert before row 1
+        LT{1, IT(1, 11, 10), true},   // Insert before row 1
+        LT{1, IT(1, 21, 11), true},   // Insert before row 4
+        LT{1, IT(1, 30, 10), true},   // Insert before row 4
+        LT{1, IT(1, 30, 20), false},  // Delete row 4
+        LT{1, IT(1, 30, 25), false},  // Delete non-existent row
+        LT{1, IT(1, 30, 30), false}   // Delete row 5
+    });
+    IdTable resultExpected = makeIdTableFromVector({
+        {10, 11},  // LT 2
+        {11, 10},  // LT 3
+        {15, 20},  // orig. Row 1
+        {15, 30},  // orig. Row 2
+        {20, 10},  // orig. Row 3
+        {21, 11},  // LT 4
+        {30, 10}   // LT 5
+    });
+
+    auto merged = locatedTriplesPerBlock.mergeTriples(1, block, 2);
+    EXPECT_THAT(merged, testing::ElementsAreArray(resultExpected));
+  }
+
+  // Merge the `LocatesTriples` into a block with 1 index columns.
+  {
+    IdTable block = makeIdTableFromVector({
+        {10},  // Row 0
+        {11},  // Row 1
+        {12},  // Row 2
+        {20},  // Row 3
+        {23},  // Row 4
+        {30}   // Row 5
+    });
+    auto locatedTriplesPerBlock = makeLocatedTriplesPerBlock({
+        LT{1, IT(1, 10, 12), false},  // Delete row 2
+        LT{1, IT(1, 10, 13), true},   // Insert before row 3
+    });
+    IdTable resultExpected = makeIdTableFromVector({
+        {10},  // orig. Row 0
+        {11},  // orig. Row 1
+        {13},  // LT 2
+        {20},  // orig. Row 3
+        {23},  // orig. Row 4
+        {30}   // orig. Row 5
+    });
+
+    auto merged = locatedTriplesPerBlock.mergeTriples(1, block, 1);
+    EXPECT_THAT(merged, testing::ElementsAreArray(resultExpected));
+  }
+
+  // Inserting a Triple that already exists should have no effect.
+  {
+    IdTable block = makeIdTableFromVector({{1, 2, 3}, {1, 3, 5}, {1, 7, 9}});
+    auto locatedTriplesPerBlock =
+        makeLocatedTriplesPerBlock({LT{1, IT(1, 3, 5), true}});
+    IdTable resultExpected = block.clone();
+
+    auto merged = locatedTriplesPerBlock.mergeTriples(1, block, 3);
+    EXPECT_THAT(merged, testing::ElementsAreArray(resultExpected));
+  }
+
+  // Inserting a Triple that already exists should have no effect.
+  {
+    IdTable block = makeIdTableFromVector({{1, 2, 3}, {1, 3, 5}, {1, 7, 9}});
+    auto locatedTriplesPerBlock = makeLocatedTriplesPerBlock(
+        {LT{1, IT(1, 2, 4), false}, LT{1, IT(1, 2, 5), false},
+         LT{1, IT(1, 3, 5), false}});
+    IdTable resultExpected = makeIdTableFromVector({{1, 2, 3}, {1, 7, 9}});
+
+    auto merged = locatedTriplesPerBlock.mergeTriples(1, block, 3);
+    EXPECT_THAT(merged, testing::ElementsAreArray(resultExpected));
+  }
+
+  // Merging if the block has additional columns.
+  {
+    IdTable block =
+        makeIdTableFromVector({{1, 2, 3, ad_utility::testing::IntId(10),
```

Comment:
bind the `IntId`lambda to a short variable (same for the UndefId), then those tests become  much more readable.